08-generic_cache/02_mutex/cache.go

```diff
@@ -0,0 +1,44 @@
+package cache
+
+import "sync"
+
+// Cache is key, value storage.
+type Cache[K comparable, V any] struct {
+	mutex sync.Mutex
```

Comment:
Might want a RWMutex

Reply:
Also, don't name the mutex "mutex". Its name should reflect what is protected. `dataMu(tex)` ?

Reply:
The general rule with regards to RWMutex is to have a demonstrable performance gain in real-world to use it. Quite often RWMutex doesn't make things faster, however, it does create quite a few potential errors e.g. using RLock instead of Lock.

Mutexes are usually named `mu` or `mtx` and the placement of mutex indicates what's being locked. Being more specific is only necessary when you have multiple mutexes. In this case, since it's for educational purpose, `mutex` or `lock` seems also appropriate.